enterprise/backup/src/main/java/org/neo4j/backup/impl/OnlineBackupCommandBuilder.java

```diff
@@ -0,0 +1,358 @@
+        return String.format( "--from=%s", address );
+    }
+
+    /**
+     * The backup location is the directory that stores multiple backups. Each directory in "backup location" is named after the backup name.
+     * In order for a backup to belong where the user wants it, the backup location is the parent of the target specified by the user.
+     * @return backup location command line argument
+     */
+    private String argBackupLocation( File targetLocation )
+    {
+        String location = Optional.ofNullable( targetLocation )
+                .map( f -> targetLocation.getParentFile() )
+                .orElseThrow( wrongArguments( "No target location specified" ) )
+                .toString();
+        return String.format( "--backup-dir=%s", location );
+    }
+
+    private String argBackupName( File targetLocation )
+    {
+        String backupName = Optional.ofNullable( targetLocation )
+                .map( File::getName )
+                .orElseThrow( wrongArguments( "No target location specified" ) );
+        return String.format( "--name=%s", backupName );
+    }
+
+    private static Supplier<IllegalArgumentException> wrongArguments( String message )
+    {
+        return () -> new IllegalArgumentException( message );
+    }
+
+    private String argFallbackToFull()
+    {
+        return Optional.ofNullable( fallbackToFull )
+                .map( flag -> String.format( "--fallback-to-full=%s", flag ) )
+                .orElse( "" );
+    }
+
+    private String argTimeout()
+    {
+        return Optional.ofNullable( this.timeout )
+                .map( value -> String.format("--timeout=%dms", value) )
+                .orElse( "" );
+    }
+
+    private String argCcOwners()
+    {
+        return Optional.ofNullable( this.consistencyCheckOwners )
+                .map( value -> String.format( "--check-consistency=%b", this.consistencyCheckOwners ) )
+                .orElse( "" );
+    }
+
+    private String argCcLabel()
+    {
+        return Optional.ofNullable( this.consistencyCheckLabel )
+                .map( value -> String.format( "--cc-label-scan-store=%b", this.consistencyCheckLabel ) )
+                .orElse( "" );
+    }
+
+    private String argCcIndexes()
+    {
+        return Optional.ofNullable( this.consistencyCheckIndexes )
+                .map( value -> String.format( "--cc-indexes=%b", this.consistencyCheckIndexes ) )
+                .orElse( "" );
+    }
+
+    private String argCcGraph()
+    {
+        return Optional.ofNullable( this.consistencyCheckGraph )
+                .map( value -> String.format( "--cc-graph=%b", this.consistencyCheckGraph ) )
+                .orElse( "" );
+    }
+
+    private String argAdditionalConf( File backupTarget )
+    {
+        if ( additionalConfig == null )
+        {
+            return "";
+        }
+        File configFile = neo4jHomeFromTarget( backupTarget ).resolve( "../additional_neo4j.conf" ).toFile();
```

Comment:
what is additional_neo4j.conf and why do we need it? how its correlate with other tools behavior?

Reply:
Can you answer this one, @phughk ?

Reply:
The idea behind this code is that you could supply an `Config additionalConfig` that would override defaults. This method in the builder would physically write that config to a file (that is hardcoded as additional_neo4j.conf) and add it as a parameter.
I don't think it is used anywhere at the moment, so if you think it should be thrown out then please go ahead.

Reply:
while the idea to have an ability to provide additional config can be a good one or not, its important that all tools that we have should have same capabilities in terms how they can be configured, where they store their logs, etc and having something that specific for a backup can be not that good idea, but i will leave it for you guys

Reply:
This feature is used in the related blockdevice PR, so let's keep it.